.github/actions/build-sign-publish-chainlink/action.yml

```
@@ -160,35 +168,66 @@ runs:
     with:
       cosign-release: 'v1.4.0'
 
-  - if: inputs.sign-images == 'true'
-    name: Sign the published root Docker image
+  - if: inputs.sign-images == 'true' && inputs.sign-method == 'keypair'
+    name: Sign the published root Docker image using keypair method
+    shell: sh
+    env:
+      COSIGN_PASSWORD: "${{ inputs.cosign-password }}"
+    run: |
+      echo "${{ inputs.cosign-private-key }}" > cosign.key
+      cosign sign --key cosign.key "${{ env.root_image_name }}"
```

After this step, would it be safer to remove the key so it never gets accidentally saved in an artifact?

```suggestion
      cosign sign --key cosign.key "${{ env.root_image_name }}"
      rm -f cosign.key
```